packages/backend-common/src/cache/types.ts

```diff
@@ -25,6 +26,21 @@ type ClientOptions = {
   defaultTtl?: number;
 };
 
+export type OptionalOnError = ((err: Error) => void) | undefined;
```

Comment:
Ehh, this should just be `export type CacheErrorHandler = (err: Error) => void;` or something; making it suport undefined is redundant on top of the `?` optionality below - especially if it's gonna be a public type